packages/trigger-sdk/src/v3/streams.ts

```
@@ -0,0 +1,512 @@
+      );
+    }
+  } catch (error) {
+    console.error(error);
+  }
+
+  const waitForStreams: Promise<void> = new Promise(async (resolve) => {
+    while (ongoingStreamPromises.length > 0) {
+      await ongoingStreamPromises.shift();
+    }
+    resolve();
+  });
```

Biome (2.1.2)

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)